File: Monitoring_Web/Monitoring_Web/ClientApp/src/app/_model/integration-api-model.ts
export interface IntegrationAPIModel {
    id: number;
    environmentId:number;
    environmentName: string | null;
    machineName: string | null;
    healthMeasurementKey: string | null;
    appid: string | null;
    domain_SystemHealth: string | null;
    nodeType: number;
    isActive: boolean | null;
    createDate: string | null;
    isCheck: boolean | null;
    isDefaultNode: boolean | null;
}export interface ListIntegrationAPI {
    pageIndex: number;
    pageSize: number;
    totalRecords: number;
    pageCount: number;
    listItem: IntegrationAPIModel[];
}
export interface NodeType{
    id: number;
    nodeType: string | null;
    description: string | null;
}export interface IntegrationAPIModelEdit {
    id: number;
    environmentID: number | null;
    machineName: string | null;
    healthMeasurementKey: string | null;
    appid: string | null;
    domain_SystemHealth: string | null;
    nodeType: number ;
    nodeTypeName: string | null;
    isActive: boolean | null;
    createDate: string | null;
    serviceList: string | null;
    isDefaultNode: boolean | null;
}

File: Monitoring_Web/Monitoring.Model/Model/IntegrationAPIModel.cs
namespace Monitoring.Model.Model
{
    public class IntegrationAPIModel
    {
        public int ID { get; set; }
        public int? EnvironmentId { get; set; }
        public string? EnvironmentName { get; set; }
        public string? MachineName { get; set; }
        public string? HealthMeasurementKey { get; set; }
        public string? Appid { get; set; }
        public string? domain_SystemHealth { get; set; }
        public Int16? NodeType { get; set; }
        public bool? IsActive { get; set; }
        public DateTime? CreateDate { get; set; }
        public bool IsCheck { get; set; }
        public bool? IsDefaultNode { get; set; }
    }
    public class IntegrationAPIModelEdit
    {
        public int ID { get; set; }
        public int? EnvironmentID { get; set; }
        public string? MachineName { get; set; }
        public string? HealthMeasurementKey { get; set; }
        public string? Appid { get; set; }
        public string? domain_SystemHealth { get; set; }
        public Int16? NodeType { get; set; }
        public bool? IsActive { get; set; }
        public DateTime? CreateDate { get; set; }
        public string? ServiceList { get; set; }
        public bool? IsDefaultNode { get; set; }
    }
}
